WasatchVCPPLib/WasatchVCPPLib/Spectrometer.cpp

```diff
@@ -298,46 +295,38 @@ int WasatchVCPP::Spectrometer::getModEnabled(void){
     return getRes[0];
 }
 
-bool WasatchVCPP::Spectrometer::getModPeriodus(void) {
+// currently for debugging
+unsigned long long WasatchVCPP::Spectrometer::getModPeriodus(void) {
     vector<uint8_t> getRes = getCmd(0xcb, 5);
     logger.debug("got modulation period. response is");
     for (auto i : getRes)
         logger.debug("%d", i);
-    return true;
+    unsigned long long modPeriod = getRes[2] >> 32 | getRes[1] >> 16 | getRes[0];
+    return modPeriod;
 }
 
 bool WasatchVCPP::Spectrometer::setModPeriodus(int us) {
     uint16_t lsw;
     uint16_t msw;
-    uint16_t* bit_buf;
-    bit_buf = to40bit(int(us));
-    lsw = bit_buf[0];
-    msw = bit_buf[1];
-    uint8_t buf[8] = { (uint8_t)bit_buf[2], 0, 0, 0, 0, 0, 0, 0 };
+    //uint16_t* bit_buf;
+    Uint40 bit_buf = Uint40(us);
+    //bit_buf = to40bit(us);
+    lsw = bit_buf.LSW;
+    msw = bit_buf.MidW;
+    uint8_t buf[8] = { (uint8_t)bit_buf.MSB, 0, 0, 0, 0, 0, 0, 0 };
     auto bytesWritten = sendCmd(0xc7, lsw, msw, buf, sizeof(buf)/sizeof(buf[0]));

```

Comment:
This is good.  Obviously you could make it shorter now by removing the "local" lsw and msw altogether, but this is fine.